.github/workflows/sync-module-tags.yaml

```diff
@@ -0,0 +1,35 @@
+on:
+  push:
+    tags: ['v*']
+
+permissions:
+  contents: write
+
+# When a repository has multiple Go modules, in order to version the non-root
+# modules, the repo must have a tag named like '<module>/<version>'. This
+# workflow responds to pushes of tags named like "vXXX" by also tagging
+# <module>/<version> accordingly, for all sub-Go-modules.
+# 
+# For example, when tagging v2.3.4, this workflow will tag the same commit with
+# pkg/signature/kms/aws/v2.3.4, so that the kms/aws module is also tagged
+# v2.3.4.
+
+jobs:
+  sync-tags:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        module:
+          - pkg/signature/kms/aws
+          - pkg/signature/kms/azure
+          - pkg/signature/kms/gcp
+          - pkg/signature/kms/hashivault
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          tag="${{ matrix.module }}/${{ github.ref_name }}"
```

Comment:
Just to check, it's `/`, not `@`, which will create the versioned module?

Reply:
That's my understanding, yes, based on [this](https://go.dev/ref/mod#vcs-version):

> If a module is defined in a subdirectory within the repository, that is, the [module subdirectory](https://go.dev/ref/mod#glos-module-subdirectory) portion of the module path is not empty, then each tag name must be prefixed with the module subdirectory, followed by a slash. For example, the module golang.org/x/tools/gopls is defined in the gopls subdirectory of the repository with root path golang.org/x/tools. The version v0.4.0 of that module must have the tag named gopls/v0.4.0 in that repository.

The [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go) repo is another example:

- the [root go.mod](https://github.com/open-telemetry/opentelemetry-go/blob/main/go.mod) corresponds to go.opentelemetry.io/otel (they use a vanity importpath, to confuse things)
- there's also [`metric/go.mod`](https://github.com/open-telemetry/opentelemetry-go/blob/main/metric/go.mod), with a tag [`metric/v0.38.1`](https://github.com/open-telemetry/opentelemetry-go/releases/tag/metric/v0.38.1) which makes that [module available at v0.38.1](https://pkg.go.dev/go.opentelemetry.io/otel/metric@v0.38.1)

Reply:
Thanks for checking!